submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+class Organism {
+    constructor(name, gender, saying, legs, hands, tail) {
+        this.name = name;
+        this.gender = gender;
+        this.species = this.constructor.name.toLowerCase();
+        this.saying = saying;
+        this.legs = legs;
+        this.hands = hands;
+        this.tail = tail;
+        this.friends = [];
+    }
+
+    addFriends(...arrayOfFriends) {
+        arrayOfFriends.forEach(({ name }) => this.friends.push(name));
+    }
+
+    description() {
+        let descriptionArray = [this.name, this.gender, this.species, this.saying, this.legs, this.hands, this.tail];
+        this.friends.length === 0 ? descriptionArray.push('none') : descriptionArray.push(this.friends);
+        return descriptionArray;
+    }
+}
+
+class Dog extends Organism {
+    static saying = 'Woof-Woof!';
+    constructor(name, gender) {
+        super(name, gender, Dog.saying, 4, 0, 1);
+    }
+}
+
+class Cat extends Organism {
+    static saying = 'Meow!';
+    constructor(name, gender) {
+        super(name, gender, Cat.saying, 4, 0, 1);
+    }
+}
+
+class Human extends Organism {
+    static saying = 'Cogito ergo sum!';
+    constructor(name, gender, saying) {
+        saying ? super(name, gender, saying, 2, 2, 0) : super(name, gender, Human.saying, 2, 2, 0);
```

Comment:
```suggestion
        super(name, gender, saying || Human.saying, 2, 2, 0);
```

or even more obvious (as default values are still worth using)

```
constructor(name, gender, saying = 'Cogito ergo sum!') {
  super(name, gender, saying, 2, 2, 0);
}
```
and you won't need `static saying`.
Perhaps, super's constructor signature will change as humans have "zero" tails, while actually do not have any.

Reply:
Done.  It's clever approach. Thanks for it.
I decide not remove static properties, it is not necessary, but in my mind, it gives useful access to saying property not only for class instance.

Reply:
A note: Ternary operator is not a replacement for `if...else`. It can be used exactly in a place you need.
So `saying ? super(name, gender, saying, 2, 2, 0) : super(name, gender, Human.saying, 2, 2, 0)` at least translates into `super(name, gender, saying ? saying : Human.saying, 2, 2, 0)`. The latter then makes `saying || Human.saying` pretty obvious.